src/Eig/EigSolver.C

```diff
@@ -190,6 +190,20 @@ static void _eig_ax(const SystemMatrix* A, const double* x, double* y)
 {
   if (!A) return;
 
+#if HAS_PETSC
+  const PETScMatrix* pA = dynamic_cast<const PETScMatrix*>(A);
+  if (pA) {
+    PETScVector X(pA->getAdm(), x, A->dim());
+    X.endAssembly();
+    PETScVector Y(pA->getAdm(), A->dim());
+    pA->multiply(X, Y);
+    PetscScalar* yv;
+    VecGetArray(Y.getVector(), &yv);
+    memcpy(y,yv,Y.dim()*sizeof(double));
```

Comment:
Looks like a memory leak to me. Where is `yv` released?

Reply:
yv is owned by Y, we borrow a reference to its internal buffer.